sdk/cosmos/azure-spring-data-cosmos/README.md

```diff
@@ -1,25 +1,356 @@
+```
+
+- Supports Optimistic Locking for specific collections, which means upserts/deletes by document will fail with an exception in case the document was modified by another process in the meanwhile. To enable Optimistic Locking for a collection, just create a string `_etag` field and mark it with the `@Version` annotation. See the following:
+
+<!-- embedme src/samples/java/com/azure/cosmos/MyDocument.java#L14-L20 -->
+```java
+@Document(collection = "myCollection")
+public class MyDocument {
+    String id;
+    String data;
+    @Version
+    String _etag;
+}
+```
+- Supports [Azure Cosmos DB partition](https://docs.microsoft.com/azure/cosmos-db/partition-data). To specify a field of domain class to be partition key field, just annotate it with `@PartitionKey`. When you do CRUD operation, pls specify your partition value. For more sample on partition CRUD, pls refer to [test here](./src/test/java/com/microsoft/azure/spring/data/cosmosdb/repository/integration/AddressRepositoryIT.java)
+- Supports [Spring Data custom query](https://docs.spring.io/spring-data/commons/docs/current/reference/html/#repositories.query-methods.details) find operation, e.g., `findByAFieldAndBField`
+- Supports [Spring Data pagable and sort](https://docs.spring.io/spring-data/commons/docs/current/reference/html/#repositories.special-parameters).
+  - Based on available RUs on the database account, cosmosDB can return documents less than or equal to the requested size.
+  - Due to this variable number of returned documents in every iteration, user should not rely on the totalPageSize, and instead iterating over pageable should be done in this way.  
+<!-- embedme src/samples/java/com/azure/cosmos/PageableRepositoryCodeSnippet.java#L29-L36 -->
+```java
+final CosmosPageRequest pageRequest = new CosmosPageRequest(0, pageSize, null);
+Page<T> page = repository.findAll(pageRequest);
+List<T> pageContent = page.getContent();
+while (page.hasNext()) {
+    Pageable nextPageable = page.nextPageable();
+    page = repository.findAll(nextPageable);
+    pageContent = page.getContent();
+}
+```
+- Supports [spring-boot-starter-data-rest](https://projects.spring.io/spring-data-rest/).
+- Supports List and nested type in domain class.
+- Configurable ObjectMapper bean with unique name `cosmosdbObjectMapper`, only configure customized ObjectMapper if you really need to. e.g.,
+<!-- embedme src/samples/java/com/azure/cosmos/ObjectMapperConfigurationCodeSnippet.java#L17-L20 -->
+```java
+@Bean(name = "cosmosdbObjectMapper")
+public ObjectMapper objectMapper() {
+    return new ObjectMapper(); // Do configuration to the ObjectMapper if required
+}
+```
 
 ## Quick Start
 
-## Snapshots
+### Add the dependency
+`spring-data-cosmosdb` is published on Maven Central Repository.  
+If you are using Maven, add the following dependency. 
+
+[//]: # "{x-version-update-start;com.microsoft.azure:spring-data-cosmosdb;current}"
+```xml
+<dependency>
+    <groupId>com.microsoft.azure</groupId>
```

Comment:
Same here, please update the group id and artifact id to the latest values which are : 

groupId -> `com.azure`
artifactId -> `azure-spring-data-cosmos`